### PR TITLE
[rllib] Basic regression tests on CartPole

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This is first cut at pre-release regression tests, currently just CartPole.

## Related issue number

https://github.com/ray-project/ray/issues/1530

### CODE REVIEW

**Comment by richardliaw** [comment]
We should outline on #1530 perhaps what we want the full regression suite to look like.

File: python/ray/rllib/tuned_examples/run_regression_tests.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from ray.tune import run_experiments
+
+
+if __name__ == '__main__':
+    experiments = {}
+
+    for test in glob.glob("regression_tests/*.yaml"):
+        config = yaml.load(open(test).read())
+        for trial in config.values():
+            trial["stop"]["training_iteration"] = 1
```
Comment (richardliaw) [inline python/ray/rllib/tuned_examples/run_regression_tests.py]: ok fix

File: python/ray/rllib/dqn/dqn_evaluator.py [hunk 2]
```diff
@@ -157,8 +157,9 @@ def _step(self, global_timestep):
         return ret
 
     def stats(self):
-        mean_100ep_reward = round(np.mean(self.episode_rewards[-101:-1]), 5)
-        mean_100ep_length = round(np.mean(self.episode_lengths[-101:-1]), 5)
+        n = self.config["smoothing_num_episodes"] + 1
```
Comment (richardliaw) [inline python/ray/rllib/dqn/dqn_evaluator.py]: would be good to add a note that there's some weird thing with the last entry and we should fix

**Review by richardliaw — APPROVED** [review]

**Comment by ericl** [comment]
Test hang seems unrelated, merging.

**Pull request merged**